errbot/backends/irc.py

```diff
@@ -177,6 +190,15 @@ def leave(self, reason=None):
             reason = ""
 
         self.connection.part(self.room, reason)
+        for attempt in range(JOIN_TIMEOUT):
+            time.sleep(.5)
+            if not self.joined:
+                break
+            time.sleep(.5)
+        else:
+            self.log.error("Timeout: Could not part %s", self.room)
+            return
```

Comment:
Maybe raise an exception though, instead of silently returning? Some of the other backends raise exceptions too.